File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/domain/exception/InvalidGuideTourStateException.java
package com.dominikgaller.alpinebooking.booking.core.domain.exception;

import com.dominikgaller.alpinebooking.booking.core.domain.GuideTourStatus;

/**
 * Thrown when a state transition is attempted on a guide tour that is not in
 * the required state.
 *
 * <p>Maps to HTTP 409 Conflict at the REST boundary.
 *
 * <p>SDD: See {@code documentation/use-cases/uc05-start-tour.spec.md}, section 8.
 */
public class InvalidGuideTourStateException extends RuntimeException {

    public InvalidGuideTourStateException(final GuideTourStatus currentStatus) {
        super("Invalid state transition from state: " + currentStatus.name());
    }
}

File: app/src/main/kotlin/com/example/service/guide/core/domain/guidetour/exception/InvalidGuideTourStateException.kt
package com.example.service.guide.core.domain.guidetour.exception

import com.example.service.guide.core.domain.guidetour.GuideTourStatus

/** Thrown when a transition is attempted from a state that does not permit it. Maps to 409. */
class InvalidGuideTourStateException(
    currentStatus: GuideTourStatus,
) : RuntimeException("Invalid state transition from state: $currentStatus")
